doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+
+Specifying a replication factor greater than one will run multiple identical
+replicas of the cluster for redundancy. Any requests to the cluster will be
+transparently routed to any live replica, so the cluster can tolerate failures
+of `n - 1` replicas before it becomes unavailable.
+
+All machines in a replica run in the same availability zone (AZ). If a cluster
+is configured with multiple replicas, replicas are distributed across AZs as
+evenly as possible to ensure availability in the event of an AZ outage.
+
+In the future, we may allow users to explicitly specify the desired
+availability zones for a cluster's replicas.
+
+The list of clusters and the status of each can be observed via the new `SHOW
+CLUSTERS` statement or the new `mz_clusters` system catalog table. The exact
+output of these commands is TODO, but they will include the provisioning status
+of the EC2 instances.
+
+It is possible to create indexes in a cluster before it is provisioned, but
+attempts to `SELECT` from these indexes will fail until the cluster is
+provisioned.
+
+The size and replication factor of an existing cluster can be changed via the new
+`ALTER CLUSTER` SQL statement:
+
+```sql
+ALTER CLUSTER <name> [SIZE {xs|s|m|l|xl}] [, REPLICATION FACTOR <n>];
+```
+
+Changing either of these properties is a zero-downtime operation that happens
+asynchronously. As with creating a cluster, users can monitor the status of an
+`ALTER CLUSTER` operation via the `SHOW CLUSTERS` statement.
+
+In the (far) future, we plan to build "autoscaling", in which users would
+specify a minimum and maximum size for their cluster, and Materialize Platform
+would automatically shrink or grow the cluster's size within those limits in
+response to observed resource utilization.
+
+Clusters are dropped via a new `DROP CLUSTER` SQL statement:
+
+```sql
+DROP CLUSTER <name> [CASCADE];
+```
+
+Clusters cannot be dropped while they contain any indexes unless `CASCADE`
+is specified.
+
+The active cluster for a session is configured via a new `cluster` session
+variable:
+
+```sql
+SET cluster = <name>;
+```
+
+Users can configure a default cluster for their sessions. How this default is
+configured is TODO.
+
+Access to a cluster is gated by SQL permission grants. Users cannot execute
+queries on a cluster on which they do not have `USAGE` privileges. Users cannot
+create clusters if they do not have the `CREATE CLUSTER` privilege in the
+region.
+
+### Database
+
+A database is a simple container of schemas that can be subject to access
+control. It confers no additional semantics.
+
+The only supported privilege is `CREATE`, which allows users to create
+schemas in the database.
+
+TODO: should databases be associated with a single [timeline](#timeline)?
+
+### Schema
+
+A schema is a simple container of database objects that can be subject to access
+control. It confers no additional semantics.
+
+The only supported privilege is `CREATE`, which allows users to create objects
```

Comment:
does create also allow alter and drop?

Reply:
No, that's reserved for the object owner, which is by default whoever created the object. SQL is weird in this regard. I have a note about this down in the "SQL access control" section.